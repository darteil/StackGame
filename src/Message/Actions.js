export const SHOW_MESSAGE = 'SHOW_MESSAGE';
export const HIDE_MESSAGE = 'HIDE_MESSAGE';

export function showMessage(text) {
  return {
    type: SHOW_MESSAGE,
    text
  };
}

export function hideMessage() {
  return {
    type: HIDE_MESSAGE
  };
}
