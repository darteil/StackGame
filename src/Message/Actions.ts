import { SHOW_MESSAGE, HIDE_MESSAGE, IShowMessage, IHideMessage } from './types';

export function showMessage(...messages: string[]): IShowMessage {
  return {
    type: SHOW_MESSAGE,
    messages
  };
}

export function hideMessage(): IHideMessage {
  return {
    type: HIDE_MESSAGE
  };
}
