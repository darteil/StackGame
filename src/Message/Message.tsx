import React, { useEffect } from 'react';
import { IMessageProps } from '.';
import styles from './styles.css';

const Message = (props: IMessageProps) => {
  let timeoutVariable: NodeJS.Timeout;

  useEffect(() => {
    timeoutVariable = setTimeout(() => {
      props.hideMessage();
    }, 10000);

    return () => clearTimeout(timeoutVariable);
  }, []);

  return (
    <div className={styles.message}>
      <em onClick={props.hideMessage} />
      <div>
        {props.messages.map((text: string, index: number) => (
          <p key={index}>{text}</p>
        ))}
      </div>
    </div>
  );
};

export default Message;
