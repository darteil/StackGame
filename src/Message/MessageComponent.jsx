import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './styles.css';

export default class MessageComponent extends Component {
  constructor(props) {
    super(props);

    this.timeoutVariable = setTimeout(() => {
      this.props.hideMessage();
    }, 10000);
  }

  componentWillUnmount() {
    clearTimeout(this.timeoutVariable);
  }

  render() {
    return (
      <div className={styles.message}>
        <em onClick={this.props.hideMessage} />
        <p>{this.props.text}</p>
      </div>
    );
  }
}

MessageComponent.propTypes = {
  text: PropTypes.string.isRequired,
  hideMessage: PropTypes.func.isRequired
};
