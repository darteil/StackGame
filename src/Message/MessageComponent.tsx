import React, { useEffect } from 'react';
import styles from './styles.css';

interface IProps {
  text: string;
  hideMessage: () => void;
}

const MessageComponent = (props: IProps) => {
  let timeoutVariable: NodeJS.Timeout;

  useEffect(() => {
    timeoutVariable = setTimeout(() => {
      props.hideMessage();
    }, 10000);

    return () => clearTimeout(timeoutVariable);
  }, []);

  return (
    <div className={styles.message}>
      <em onClick={props.hideMessage} />
      <p>{props.text}</p>
    </div>
  );
};

export default MessageComponent;
