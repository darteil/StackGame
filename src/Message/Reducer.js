import { SHOW_MESSAGE, HIDE_MESSAGE } from './Actions';

const initialState = {
  text: '',
  show: false
};

export default function Message(state = initialState, action) {
  switch (action.type) {
    case SHOW_MESSAGE: {
      return { ...state, show: true, text: action.text };
    }
    case HIDE_MESSAGE: {
      return { ...state, show: false, text: '' };
    }
    default:
      return state;
  }
}
