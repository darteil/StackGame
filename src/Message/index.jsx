import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { hideMessage } from 'App/Message/Actions';
import styles from './styles.css';

class Message extends Component {
  constructor(props) {
    super(props);

    this.timeoutVariable = setTimeout(() => { this.props.hideMessage(); }, 10000);
  }

  componentWillUnmount() {
    clearTimeout(this.timeoutVariable);
  }

  render() {
    return (
      <div className={styles.message}>
        <em onClick={this.props.hideMessage} />
        <p>{this.props.text}</p>
      </div>
    );
  }
}

Message.propTypes = {
  text: PropTypes.string.isRequired,
  hideMessage: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  text: state.Message.text
});

const mapDispatchToProps = dispatch => ({
  hideMessage: () => {
    dispatch(hideMessage());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(Message);

