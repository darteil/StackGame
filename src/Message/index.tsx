import React from 'react';
import { Dispatch } from 'redux';
import { connect } from 'react-redux';
import { hideMessage } from './Actions';
import { AppState } from '../store';
import Message from './Message';

export interface IMessageProps {
  messages: string[];
  hideMessage: () => void;
}

const MessageComponent = (props: IMessageProps) => <Message {...props} />;

const mapStateToProps = (state: AppState) => ({
  messages: state.Message.messages
});

const mapDispatchToProps = (dispatch: Dispatch) => ({
  hideMessage: () => {
    dispatch(hideMessage());
  }
});

export const MessageContainer = connect(mapStateToProps, mapDispatchToProps)(MessageComponent);
