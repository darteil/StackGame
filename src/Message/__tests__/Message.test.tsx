import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Message from '../Message';

describe('Message component', () => {
  it('should render correctly', () => {
    const text = ['text example'];
    const handleHide = jest.fn();

    const output = shallow(<Message hideMessage={handleHide} messages={text} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
