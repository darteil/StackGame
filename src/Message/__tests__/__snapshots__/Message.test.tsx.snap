// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component should render correctly 1`] = `
<div
  className="message"
>
  <em
    onClick={[MockFunction]}
  />
  <div>
    <p
      key="0"
    >
      text example
    </p>
  </div>
</div>
`;
